go/vt/vtgate/safe_session.go

```diff
@@ -40,6 +40,11 @@ type SafeSession struct {
 	mustRollback    bool
 	autocommitState autocommitState
 	commitOrder     vtgatepb.CommitOrder
+	savepointState  savepointState
+	// rollbackOnPartialExec is set if any DML was successfully
```

Comment:
set to what? why is this a string?

Reply:
rollbackOnPartialExec will store the `rollback to <savepoint>` sql query which is generated randomly when doing MarkSavepoint.

Reply:
we should add :point_up: to the comment